### PR TITLE
feat: Setup CD on develop branch

### LINKED ISSUE #10: Setup CD on develop branch
Every commit on develop branch should automatically get deployed to gh pages. Currently we do a `npm run deploy` locally to make a deployment and would like this process to be automated.

### PR DESCRIPTION
**Pull request by rishi-raj-jain**
fix #10 
/claim #10

### CODE REVIEW

**Comment by rishi-raj-jain** [comment]
Screenshot 2023-09-28 at 4 54 12 PM

How can one find this file?

**Comment by rishi-raj-jain** [comment]
I don't see anything on develop branch that has ./src/gen/tailcall-version.json.

**Comment by rishi-raj-jain** [comment]
ah okay gotcha I need to spin up a folder if it doesn't exist.

**Comment by rishi-raj-jain** [comment]
Also the build script is using the wrong order of scripts.

**Comment by rishi-raj-jain** [comment]
can you post the environment variables that you use for npm run deploy here please?

**Comment by rishi-raj-jain** [comment]
Updated the PR to have:

- Update package lock as it missed a dep
- Updated generate script as it missed an important step
- updated package.json as "build" script was using wrong order of commands

**Comment by rishi-raj-jain** [comment]
Updated the PR to have:
- onBrokenLinks as log so that it doesn't stop the build
- update the workflow to work flawlessly

File: package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "scripts": {
     "docusaurus": "docusaurus",
     "start": "npm run generate; docusaurus start",
-    "build": "docusaurus build; npm run generate",
+    "build": " npm run generate; docusaurus build",
```
Comment (tusharmath) [inline package.json:8]:
```suggestion
    "build": "npm run generate; docusaurus build",
```

File: docusaurus.config.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ const config = {
   tagline: "<tagline>",
   url: "https://tailcall.run",
   baseUrl: "/",
-  onBrokenLinks: "throw",
+  onBrokenLinks: "log",
```
Comment (tusharmath) [inline docusaurus.config.js:15]: i think we should throw here.

**Comment by tusharmath** [comment]
Is there a way to test this without merging to main?

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Install and Deploy 🔧
+        run: |
+          npm install
+          npm run build
+
+      - name: Deploy 🚀
```
Comment (tusharmath) [inline .github/workflows/deploy.yml:24]: why not use `npm run deploy`?

Comment (rishi-raj-jain) [reply .github/workflows/deploy.yml:24]: docusarus deploy requires some env variables. why not just go with build and deploy?

**Comment by rishi-raj-jain** [comment]
Screenshot 2023-09-28 at 5 16 45 PM

@tusharmath 

Right now the existing links throw error. Can you resolve this on your end?

**Comment by tusharmath** [comment]
> Is there a way to test this without merging to main?

thoughts on this?

**Comment by rishi-raj-jain** [comment]
> thoughts on this?

ah yes, please have a look at https://github.com/rishi-raj-jain/tailcallhq.github.io/actions/runs/6338397551/job/17215416472. basically, I'm deploying to my fork to test with.

**Comment by rishi-raj-jain** [comment]
https://github.com/rishi-raj-jain/tailcallhq.github.io/actions is where you can find all the runs.

**Comment by rishi-raj-jain** [comment]
> Screenshot 2023-09-28 at 5 16 45 PM
> @tusharmath
> 
> Right now the existing link in markdown throw error to the url it's linking to. Can you resolve this on your end?

That's why I had changed it the brokenLinks flag to "log" so that such logs are visible but don't break the build/

**Comment by tusharmath** [comment]
Fixed, take a pull from `develop`

**Comment by rishi-raj-jain** [comment]
Awesome that worked, now eyes on https://github.com/rishi-raj-jain/tailcallhq.github.io/actions.

**Comment by rishi-raj-jain** [comment]
@tusharmath 

- https://github.com/rishi-raj-jain/tailcallhq.github.io/actions/runs/6338480891 passed.
- https://github.com/rishi-raj-jain/tailcallhq.github.io/tree/gh-pages was created with the build files that are deployed.
- With the configuration on the repo as (basically set to deploy from gh-pages):
Screenshot 2023-09-28 at 5 28 57 PM this shall work.

**Review by tusharmath — APPROVED** [review]

**Pull request merged**